canopy_utils.ml

```diff
@@ -34,18 +34,17 @@ let ptime_to_pretty_date t =
   Ptime.to_date t |> fun (y, m, d) ->
     Printf.sprintf "%04d-%02d-%02d" y m d
 
-module KeyHashtbl = struct
-  module KeyHash = struct
+module KeyMap = struct
+  module KeyOrd = struct
     type t = string list
-    let equal = (=)
-    let hash = Hashtbl.hash
+    let compare = Pervasives.compare
```

Comment:
I wouldn't rely on `Pervasives.compare` (but maybe it is fine here).. I always struggle whether it does the right thing or not.  I'd state it explicitly:
```OCaml
let compare a b = match compare (List.length a) (List.length b) with
  | 0 -> (try List.find ((<>) 0) (List.map2 String.compare a b) with Not_found -> 0)
  | x -> x
```